lib/mri/mkmf.rb

```diff
@@ -598,7 +598,7 @@ def try_compile(src, opt="", *opts, &b)
   # [+opt+] a String which contains preprocessor options
   def try_cpp(src, opt="", *opts, &b)
     try_do(src, cpp_command(CPPOUTFILE, opt), *opts, &b) and
-      File.file?("#{CONFTEST}.i")
+      File.file?("#{CONFTEST}.bc") # Truffle: Update ext from .i to .bc
```

Comment:
I think the problem is somewhere else:
```
ruby -e 'p RbConfig::CONFIG["CPPOUTFILE"]'   
"-o conftest.i"

jt ruby -e 'p RbConfig::CONFIG["CPPOUTFILE"]'
nil
```
Could you try defining CPPOUTFILE in RbConfig instead?